staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -228,16 +240,19 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			string(types.MergePatchType),
 		}
 		handler := handlers.PatchResource(storage, requestScope, r.admission, unstructured.UnstructuredObjectConverter{}, supportedTypes)
+		handler = metrics.InstrumentHandlerFunc(verb, resource, subresource, scope, handler)
 		handler(w, req)
 		return
 	case "delete":
 		allowsOptions := true
 		handler := handlers.DeleteResource(storage, allowsOptions, requestScope, r.admission)
+		handler = metrics.InstrumentHandlerFunc(verb, resource, subresource, scope, handler)
 		handler(w, req)
 		return
 	case "deletecollection":
 		checkBody := true
 		handler := handlers.DeleteCollection(storage, checkBody, requestScope, r.admission)
+		handler = metrics.InstrumentHandlerFunc(verb, resource, subresource, scope, handler)
```

Comment:
These metrics.InstrumentHandlerFunc calls seem duplicated, maybe we can just set the handler in each of these blocks and call this once?

Reply:
with a global handler var, we could remove the instrumentation call, the `handler(w,req)` call and the return and do those uniformely for all non-default cases below the switch.

Reply:
Done.